modules/apps/headless/headless-delivery/headless-delivery-test/src/testFunctional/macros/DocumentFolderAPI.macro

```diff
@@ -76,4 +165,48 @@ definition {
 		return ${response};
 	}
 
+	macro getDocumentFoldersWithDifferentPameter {
+		var curl = DocumentFolderAPI._curlDocumentFolders(
+			groupName = ${groupName},
+			parameter = ${parameter},
+			parameterValue = ${parameterValue});
+
+		var response = JSONCurlUtil.get(${curl});
+
+		return ${response};
+	}
+
+	macro setUpGlobalDocumentFolder {
+		Variables.assertDefined(parameterList = ${groupName});
+
+		var response = DocumentFolderAPI.addDocumentFolder(
+			externalReferenceCode = ${externalReferenceCode},
+			groupName = ${groupName},
+			name = ${name});
+
+		var documentFolderId = JSONPathUtil.getIdValue(response = ${response});
+		var externalReferenceCode = JSONPathUtil.getErcValue(response = ${response});
+		static var staticDoucmentFolderId = ${documentFolderId};
+		static var staticDoucmentFolderErc = ${externalReferenceCode};
+
+		return ${staticDoucmentFolderId};
+
+		return ${staticDoucmentFolderErc};
+	}
+
+	macro updateDocumentFolderBySiteIdAndErc {
```

Comment:
if we update by SiteId and Erc then it means that these parameters are required? If so, they need to be put in Variables.assertDefined

Reply:
Sure, I will modify it to "Variables.assertDefined(parameterList = "${groupName},${externalReferenceCode}");"